app/code/Magento/InventoryIndexer/Model/GetStockItemData.php

```diff
@@ -50,6 +51,13 @@ public function execute(string $sku, int $stockId)
             ->from($stockItemTableName, [IndexStructure::QUANTITY, IndexStructure::IS_SALABLE])
             ->where(IndexStructure::SKU . ' = ?', $sku);
 
-        return $connection->fetchRow($select) ?: null;
+        try {
+            return $connection->fetchRow($select) ?: null;
```

Comment:
Need to cover this functionality with integration test

Reply:
@naydav done.